nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -183,8 +210,12 @@ WaypointFollower::followWaypoints()
       }
     } else if (current_goal_status_ == ActionStatus::SUCCEEDED) {
       RCLCPP_INFO(
-        get_logger(), "Succeeded processing waypoint %i, "
-        "moving to next.", goal_index);
+        get_logger(), "Succeeded processing waypoint %i, processing waypoint task execution",
+        goal_index);
+      auto node = shared_from_this();
+      waypoint_task_executor_->processAtWaypoint(goal->poses[goal_index], goal_index);
```

Comment:
Oh here's a good idea, we should change this API so that `processAtWaypoint` can return a `bool` and we can check if its `false` to also treat it as a fail checking on `stop_on_failure_` for whether to exit or continue to the next one

Reply:
nice idea, I changed the return type of `processAtWaypoint` to `bool`, but if the plugin is disabled or `waypoint_pause_duration` is set to zero, we will always have to return `false` from `processAtWaypoint` call, right ? that will make `stop_on_failure_` to be depending on the plugin to be enabled and active, not sure if that is behavior want ?

Reply:
If it fails to process & `stop_on_failure_` is true, we should exit with failure. 

I think maybe just not enabled returns true since its not a "failure", its just not turned on.

Reply:
I will address this after we confirm whether there will be a follow up PR